_pages/documentation/CPU-Models/TraceCPU.md

```diff
@@ -0,0 +1,159 @@
+repeated uint64 reg_dep &nbsp;              | Past instruction number on which there is RAW data dependency
+optional uint32 weight &nbsp; | 	To account for committed instructions that were filtered out
+optional uint64 pc &nbsp; | Instruction address, i.e. the program counter
+optional uint64 v_addr &nbsp; | 	Virtual memory address if instruction is a load/store
+optional uint32 asid &nbsp; | Address Space ID
+
+A decode script in Python is available at `util/decode_inst_dep_trace.py` that outputs the trace in ASCII format.
+
+**Example of a trace in ASCII**
+
+1,356521,COMP,8500::
+
+2,35656,1,COMP,0:,1:
+
+3,35660,1,LOAD,1748752,4,74,500:,2:
+
+4,35660,1,COMP,0:,3:
+
+5,35664,1,COMP,3000::,4
+
+6,35666,1,STORE,1748752,4,74,1000:,3:,4,5
+
+7,35666,1,COMP,3000::,4
+
+8,35670,1,STORE,1748748,4,74,0:,6,3:,7
+
+9,35670,1,COMP,500::,7
+
+Each record in the instruction fetch trace has the following fields.
+
+Fields    | Discritption
+-------------- | -------------
+required uint64 tick    |	Timestamp of the access
+required uint32 cmd	    | Read or Write (in this case always Read)
+required uint64 addr	| Physical memory address
+required uint32 size	| Size in bytes of data
+optional uint32 flags	| Flags or attributes of the access
+optional uint64 pkt_id &nbsp;  |   Id of the access
+optional uint64 pc      |	Instruction address, i.e. the program counter
+
+
+
+The decode script in Python at `util/decode_packet_trace.py` can be used to output the trace in ASCII format.
+
+
+**Compile dependencies**:
+
+You need to install google protocol buffer as the traces are recorded using this.
+
+```js
+sudo apt-get install protobuf-compiler
+sudo apt-get install libprotobuf-dev
+```
+
+### **Scripts and options**
+* SE mode
+    * `build/ARM/gem5.opt [gem5.opt options] -d bzip_10Minsts configs/example/se.py [se.py options] --cpu-type=arm_detailed --caches --cmd=$M5_PATH/binaries/arm_arm/linux/bzip2 --options=$M5_PATH/data/bzip2/lgred/input/input.source -I 10000000 --elastic-trace-en --data-trace-file=deptrace.proto.gz --inst-trace-file=fetchtrace.proto.gz --mem-type=SimpleMemory`
+* FS mode: Create a checkpoint for your region of interest and resume from the checkpoint but with O3 CPU model and tracing enabled
+    * `build/ARM/gem5.opt --outdir=m5out/bbench ./configs/example/fs.py [fs.py options] --benchmark bbench-ics`
+    * `build/ARM/gem5.opt --outdir=m5out/bbench/capture_10M ./configs/example/fs.py [fs.py options] --cpu-type=arm_detailed --caches --elastic-trace-en --data-trace-file=deptrace.proto.gz --inst-trace-file=fetchtrace.proto.gz --mem-type=SimpleMemory --checkpoint-dir=m5out/bbench -r 0 --benchmark bbench-ics -I 10000000`
+  
+
+## **Replay with Trace CPU**
+
+The execution trace generated above is then consumed by the Trace CPU as illustrated below.
+
+![Trace_cpu_top_level](../../../assets/img/Trace_cpu_top_level.jpg)
```

Comment:
Let's not make this link relative, but absolute. It would be better to remove the '../../../'